lucene/core/src/java/org/apache/lucene/util/automaton/Automata.java

```diff
@@ -85,7 +85,22 @@ public static Automaton makeAnyBinary() {
     a.finishState();
     return a;
   }
-  
+
+  /**
+   * Returns a new (deterministic) automaton that accepts all binary terms except
+   * the empty string.
+   */
+  public static Automaton makeAnyBinaryExceptEmpty() {
+    Automaton a = new Automaton();
+    int s1 = a.createState();
+    int s2 = a.createState();
+    a.setAccept(s2, true);
```

Comment:
Do we need to call setAccept on s1 too? Or is it done implicitly elsewhere (I'm not completely familiar with the Automaton code)

Reply:
Oops ignore this comment I was totally confused. :)